src/mlpack/core/util/arma_traits.hpp

```diff
@@ -217,6 +217,14 @@ struct GetDenseMatType<arma::SpMat<eT>>
   typedef arma::Mat<eT> type;
 };
 
+// Get the cube type corresponding to a given MatType.
+
+template<typename MatType>
+struct GetCubeType
+{
+  typedef arma::Cube<typename MatType::elem_type> type;
+};
```

Comment:
This will work for any given `MatType`, which seems unsafe.  Like I mentioned in the last comment I would suggest using an empty struct for non-Armadillo types:

```
template<typename MatType>
struct GetCubeType;

template<typename eT>
struct GetCubeType<arma::Mat<eT>>
{
  typedef arma::Cube<eT> type;
};
```

If you want, we can handle this in a later PR, but in my experience it's almost always better to solve the problem sooner rather than later.  In this particular case I could see it either way.

Reply:
I will take care of it, no worries